Restructring alerts page

sangupta1 opened the pull request.

makambalaji committed:
Consolidate plugins/index.js file in cypress framework
updated app.ts and index.js files

sangupta1 committed:
Added logs for the Alert Pod on Monitring/Alerts page

glekner committed:
converting yaml-templates extension types to dynamic-plugin-sdk

sangupta1 committed:
restructured alerts page v2

sangupta1 committed:
sample-commit

sangupta1 committed:
restructred monitoring alerts page

sangupta1 committed:
rebased with upstream and structured alerts page

nemesis09 committed:
pipelinerun visualisation for status PipelineRunPending

divyanshiGupta commented on frontend/public/components/factory/details.tsx:
```diff
@@ -164,6 +173,12 @@ export type DetailsPageProps = {
+  loadError?: string;
+  rule?: Rule;
+  silencesLoaded?: boolean;
+  ele?: JSX.Element;
```
@sangupta1 why do you need to change details component thereby also changing PageHeading and HorizontalNav?

Ideally we should not change it for a specific resource as this is a generic component used for other resources as well

sangupta1 replied:
@divyanshiGupta I needed to change the Details component because the PageHeading for Alert comprises of its Monitoring Resource Icon, Severity and 'blue' Silence Alert Button. However, before i changed the PageHeading component, it comprised of showing the Resource status and a 'white' Actions button. So, in order to adjust the changes required for the heading on ALerts page, i changed it. 
![Screenshot from 2021-04-18 23-33-33](https://user-images.githubusercontent.com/77098226/115155674-9e893080-a09e-11eb-8454-6c72acb9044d.png)
![Screenshot from 2021-04-18 23-34-56](https://user-images.githubusercontent.com/77098226/115155709-b365c400-a09e-11eb-88d9-40503a7c9657.png)

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -585,180 +588,222 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
+        //{...props}
+        //getResourceStatus={podPhase}
+        //menuActions={menuActions}
+        pages={[navFactory.details(Details), navFactory.logs(AlertLogs)]}
```
Instead of refactoring DetailsPage you can pass params to these pages like this 
```suggestion
        pages={[navFactory.details(() => <Details alert={alert} {...otherProps} />), navFactory.logs(AlertLogs)]}
```

sangupta1 replied:
Thank you. I will make this change and this will also enable me not to change the HorizontalNav component.

sangupta1 replied:
@divyanshiGupta I have made the necessary changes as suggested by you . Should i raise another PR for it after closing this one?

divyanshiGupta replied:
@sangupta1 No don't need to raise another PR. I will try to review it today. 

Also the same alerts-page is used in devconsole monitoring so make sure that also works as expected. I will also try to run the code today. Let me know if additional setup steps are involved to see the logs.

sangupta1 replied:
Thank you. No additional setup is required as of now to see the logs. Currently, All the logs corresponding to the pod and namespace affected by the alert are displayed there.  I will check if things work as expected on the devconsole monitoring.

sangupta1 replied:
@divyanshiGupta Things are working as expected on devconsole monitoring as well. However, one issue i have experienced is that on the Alert Detail page for a particular alert, when switching from 'Details' tab to 'Logs' tab or vice-versa, then the corresponding component is not rendered. This issue is in both Administrator and Developer view on the console while switching the tabs. I was unable to resolve this.

divyanshiGupta replied:
Okay. I will try to run your code today and see why that is happening and will let you know. After that issue is resolved we can raise a PR on upstream console.

sangupta1 replied:
Thanks a lot @divyanshiGupta

sangupta1 committed:
removed changes in horizontalNav

debsmita1 committed:
update devfile form for preselected devfile

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -585,180 +588,222 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
+          },
+          { name: t('public~Alert details'), path: undefined },
+        ]}
+        pages={[
+          navFactory.details(() => (
+            <Details
+              alert={alert}
+              rule={rule}
+              namespace={alertPodObj.namespace}
+              silencesLoaded={silencesLoaded}
+            />
+          )),
+          navFactory.logs(AlertLogs),
+        ]}
```
```suggestion
         pages={[
          {
            href: '',
            queryParams,
            // t('details-page~Details')
            nameKey: 'details-page~Details',
            component: Details,
            pageData: {
              alert,
              rule,
              namespace: alertPodObj.namespace,
              silencesLoaded,
            },
          },
          {
            href: 'logs',
            queryParams,
            // t('details-page~Logs')
            nameKey: 'details-page~Logs',
            component: AlertLogs,
          },
        ]}
```

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -1712,6 +1757,7 @@ type AlertsDetailsPageProps = {
+  match?: any;
 };
 
 type AlertMessageProps = {
```
The issue why your component was not getting rendered is because `exact` is added to the route and which means the url path and route path on which the component is supposed to render should be exactly same 
https://github.com/divyanshiGupta/console/blob/7b41f1bda5f31470cf229ef1aab7d7ec657c147e/frontend/public/components/monitoring/alerting.tsx#L1732
Remove `exact` from here

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -585,180 +588,222 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
+          )),
+          navFactory.logs(AlertLogs),
+        ]}
+      />
```
Also `queryParams` are required for fetching the alert and these were getting lost on clicking the tabs as horizontal-nav doesnt have support for adding `queryParams`.

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -1712,6 +1757,7 @@ type AlertsDetailsPageProps = {
   namespace: string;
   rule: Rule;
   silencesLoaded: boolean;
+  match?: any;
 };
```
```suggestion
  match?: any;
  location? Location<any>;
};
```

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -585,180 +588,222 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
   };
```
`const alertStateToProps = (state: RootState, { match, location }): AlertsDetailsPageProps `
```suggestion
location,
  };
```

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -585,180 +588,222 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
 export const AlertsDetailsPage = withFallback(
   connect(alertStateToProps)((props: AlertsDetailsPageProps) => {
-    const { alert, loaded, loadError, namespace, rule, silencesLoaded } = props;
+    const { alert, loaded, loadError, namespace, rule, silencesLoaded, match } = props;
```
```suggestion
    const { alert, loaded, loadError, namespace, rule, silencesLoaded, match, loaction } = props;
```

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -585,180 +588,222 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
     const labelsMemoKey = JSON.stringify(alert?.labels);
     // eslint-disable-next-line react-hooks/exhaustive-deps
     const labels: PrometheusLabels = React.useMemo(() => alert?.labels, [labelsMemoKey]);
 
```
```suggestion
 const queryParams: string = React.useMemo(() => location?.search, []);
```

divyanshiGupta commented:
To add queryParams support to horizontal-nav do following
https://github.com/divyanshiGupta/console/blob/881154954ed84ed47669b41399ecc3a0930b5b09/frontend/public/components/utils/horizontal-nav.tsx#L52
Add `queryParams?: string` in `Page`

https://github.com/divyanshiGupta/console/blob/881154954ed84ed47669b41399ecc3a0930b5b09/frontend/public/components/utils/horizontal-nav.tsx#L184
`to={`${baseURL.replace(/\/$/, '')}/${href}${queryParams ?? ''}`}`

divyanshiGupta commented:
Also in MonitoringAlertsDetailsPage send location param to AlertsDetailsPage to persist queryParams for devconsole alerts as well.

divyanshiGupta commented:
@sangupta1  With these changes in logs page heading and AlertLogs component is getting rendered but there is some issue in the component as it is not getting resource obj and hence the logs are not shown. Try to find out the reason behind this and also covert your AlertLogs component into a FunctionalComponent and use useState hook for setting the state.

sangupta1 commented:
Thank you @divyanshiGupta . I will make these changes and raise a PR to your repo.

divyanshiGupta commented:
@sangupta1 push the changes in this PR only.

sangupta1 commented:
> @sangupta1 With these changes in logs page heading and AlertLogs component is getting rendered but there is some issue in the component as it is not getting resource obj and hence the logs are not shown. Try to find out the reason behind this and also covert your AlertLogs component into a FunctionalComponent and use useState hook for setting the state.

@divyanshiGupta With the changes that you suggested, the logs are getting rendered properly for those alerts that have pod and namespace associated with them. Only, for those alerts which don't have pod and namespace associated with them, for them no logs are shown. This was the desired functionality. I will add a message "No logs for this Alert" for such alerts. Also, the AlertLogs component is a copy of PodLogs component in pod-logs.jsx file with some minor changes. This PodLogs component is a class component. Should i still convert the AlertLogs into a functional component?

rottencandy committed:
fill & hide sample app for preselected devfile

rottencandy committed:
create item provider for devfile samples

sangupta1 committed:
fixed navigation issue on alerts page

invincibleJai committed:
adds devfile provider for catalog

divyanshiGupta commented:
> > @sangupta1 With these changes in logs page heading and AlertLogs component is getting rendered but there is some issue in the component as it is not getting resource obj and hence the logs are not shown. Try to find out the reason behind this and also covert your AlertLogs component into a FunctionalComponent and use useState hook for setting the state.
> 
> @divyanshiGupta With the changes that you suggested, the logs are getting rendered properly for those alerts that have pod and namespace associated with them. Only, for those alerts which don't have pod and namespace associated with them, for them no logs are shown. This was the desired functionality. I will add a message "No logs for this Alert" for such alerts. Also, the AlertLogs component is a copy of PodLogs component in pod-logs.jsx file with some minor changes. This PodLogs component is a class component. Should i still convert the AlertLogs into a functional component?

@sangupta1 pod-log.jsx is an old component. Any new components should be FCs unless in any case its not possible which mostly is not going to be.

sangupta1 commented:
@divyanshiGupta Okay. I will convert the AlertLogs component into a functional component and let you know when i have updated the PR with this change.

dtaylor113 committed:
Wait for Cluster Auth Operator to start and finish Progressing 'test' IDP before e2e tests starts

sangupta1 committed:
converted AlertLogs into a functional component

sangupta1 committed:
corrected mistake in renaming component

sangupta1 committed:
corrected useEffect array

makambalaji committed:
epic automation for workspaces

divyanshiGupta commented on frontend/public/components/utils/horizontal-nav.tsx:
```diff
@@ -181,7 +182,7 @@ export const NavBar = withRouter<NavBarProps>(({ pages, baseURL, basePath }) =>
           <li className={klass} key={nameKey || name}>
             <Link
-              to={`${baseURL.replace(/\/$/, '')}/${href}`}
+              to={`${baseURL.replace(/\/$/, '')}/${href}${queryParams ? queryParams : ''}`}
```
```suggestion
              to={`${baseURL.replace(/\/$/, '')}/${href}${queryParams ?? ''}`}
```

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.jsx:
```diff
@@ -0,0 +1,131 @@
+    rule,
+  };
+};
+
```
You can create another file as `alert-logs-utils.ts` and move all the utilities over there.

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.jsx:
```diff
@@ -0,0 +1,131 @@
+    }
+  }, [props.obj]);
+
+  function _selectContainer(name) {
```
```suggestion
 const selectContainer = (name) => {
```

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.jsx:
```diff
@@ -0,0 +1,131 @@
+    setQueryArgument('container', currentKey);
+  }
+
+  const currentContainer = _.get(containers, currentKey) || _.get(initContainers, currentKey);
```
```suggestion
  const currentContainer = containers?.currentKey ?? initContainers?.currentKey;
```

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.jsx:
```diff
@@ -0,0 +1,131 @@
+
+  React.useEffect(() => {
+    const build = props.obj;
+    const currentContainers = _.get(build, 'spec.containers', []);
+    const currentInitContainers = _.get(build, 'spec.initContainers', []);
```
Use optional chaining

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.jsx:
```diff
@@ -0,0 +1,131 @@
+  );
+
+  return (
+    <>
+      {props.alert?.labels?.namespace && (
+        <div className="co-m-pane__body">
+          <ResourceLog
+            containerName={currentContainer ? currentContainer.name : ''}
+            dropdown={containerDropdown}
+            resource={props.obj}
+            resourceStatus={currentContainerStatus}
+          />
+        </div>
+      )}
+      {!props.alert?.labels?.namespace && (
+        <div className="co-m-pane__body">No logs for this Alert</div>
+      )}
+    </>
```
```suggestion
    <div className="co-m-pane__body">
        {props.alert?.labels?.namespace ? (
          <ResourceLog
            containerName={currentContainer ? currentContainer.name : ''}
            dropdown={containerDropdown}
            resource={props.obj}
            resourceStatus={currentContainerStatus}
          /> ) : No logs for this Alert
      )}
      </div>
```

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.jsx:
```diff
@@ -0,0 +1,131 @@
+  );
+};
+
+export default connect(alertStateToProps)(AlertLogs);
```
Use typescript here and add types wherever necessary.

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.jsx:
```diff
@@ -0,0 +1,131 @@
+    setInitContainers(containersToStatuses(build, currentInitContainers));
+    if (!currentKey) {
+      const firstContainer = _.find(containers, { order: 0 });
+      setCurrentKey(firstContainer ? firstContainer.name : '');
```
This will go into an infinite loop if firstContainer is always undefined.

sangupta1 replied:
@divyanshiGupta With the changes i pushed now, the useEffect will not go into an infinite loop even if the firstContainer is undefined.

debsmita1 committed:
give feedback when modal submit is clicked

pcbailey committed:
Convert form help uses of Popover to FieldLevelHelp

sangupta1 committed:
added typescript and alert-logs-utils file

abhinandan13jan committed:
make taskrun calls namespace scoped

maysunfaisal committed:
vendor changes

Signed-off-by: Maysun J Faisal <maysunaneek@gmail.com>

nemesis09 committed:
redesign add page to follow masonry layout

karthikjeeyar committed:
Fix tooltip on pipelineRunCount and TaskRunDuration graphs based of PF guidelines

jotak committed:
Fix documentation link to network policies

Also fix missing space in english text

i18n network-policy with link: use Trans

rexagod committed:
Run discovery directly on nodes.

Run discovery directly on the node when user clicks that node's
respective "Discover Disks" button.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

yaacov committed:
Move kubevirt API to v1

Signed-off-by: yaacov <kobi.zamir@gmail.com>

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs-utils.ts:
```diff
@@ -0,0 +1,77 @@
+  loaded: boolean;
+  loadError?: string;
+  rule: Rule;
+};
```
Prop type should be kept in the same file as the component.

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.tsx:
```diff
@@ -0,0 +1,100 @@
+          <div>No logs for this Alert</div>
+        )}
+      </div>
+    </>
```
You don't need the Fragment here.

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -582,183 +585,239 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
+    </>
+  );
+};
+
```
It will good if we can extract out this component into a separate file as it will make this file a bit smaller and more readable.

kdoberst committed:
Allow multiple selection on resources drop down

This allows the user to select multiple options without closing the drop down for the Resources drop down on the Events and Search pages.

This swapped out the custom Dropdown component with the PatternFly Select component.

Commit:
Bug 1957756: Fix key value in replacement object

- key is `node` but we are fetching `nodeName`.
- removed the check for metrics map since it's redundant.
- updated check for the replaced device based on osd, path, deviceID, serial and node combination.

Signed-off-by: Afreen Rahman <afrahman@redhat.com>

rottencandy committed:
avoid using master for builder image detection

TheRealJon committed:
Bug 1943238: Display olm conditions descriptors in their own page section on operand details pages

Update condition descriptors to be rendered in separate, full width tables like the standard status.conditions table

reginaelyse committed:
add error handling if appGroups is null

gajanan-more committed:
Topology List View Automation

GowthamShanmugam committed:
Bug 1946243: Fix pool creation timout issue

Stopping pool creation when pool CR status become Failure

Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

Commit:
Gherkin: Implement the support needed for managed services
Gherkin: Visualisation of serverless function
Gherkin: Connection from kafka source to kafka connection

andrewballantyne committed:
Global i18n mocks

andrewballantyne committed:
Pipeline i18n infra improvements

openshift-merge-robot committed:
Merge pull request #8878 from dtaylor113/cypress-projectdropdown-assertion

Bug 1957895: Cypress helper projectDropdown.shouldContain is not an assertion

openshift-merge-robot committed:
Merge pull request #8883 from florkbr/bug-1957561-cluster-settings-i18n

Bug 1957561: lack of pseudolocalization for some text on Cluster Setting

openshift-merge-robot committed:
Merge pull request #8885 from jerolimov/odc-5820

Fix that boolean values (esp. false) are not loaded and synced correctly

SanjalKatiyar committed:
Disable pool compression via UI

maysunfaisal committed:
Add Devfile Sample Endpoint

Signed-off-by: Maysun J Faisal <maysunaneek@gmail.com>

openshift-merge-robot committed:
Merge pull request #8786 from rebeccaalpert/i18n-user-management

Bug 1942715: Fix User Management i18n misses

openshift-merge-robot committed:
Merge pull request #8790 from Jayashree-panda/external

Bug 1952832: Enable external mode for IBM cloud

openshift-merge-robot committed:
Merge pull request #8855 from debsmita1/modal-submit

give feedback when modal submit is clicked

openshift-merge-robot committed:
Merge pull request #8887 from invincibleJai/fix-bz-1958439

Bug 1958439: fixes issue with html5 required validation for dynamic forms

openshift-merge-robot committed:
Merge pull request #8889 from sg00dwin/cluster-update-translation-msg

Bug 1957590: Translation of cluster updating message

openshift-merge-robot committed:
Merge pull request #8890 from divyanshiGupta/ODC-5609

Navigate user to respective catalog-page on pressing enter while focused on quick-add view-all link

openshift-merge-robot committed:
Merge pull request #8687 from maysunfaisal/sample-1

Add Devfile Sample Endpoint

openshift-merge-robot committed:
Merge pull request #8897 from SanjalKatiyar/pool_compression_mode

Bug 1958679: Disable pool compression via UI

gouyang committed:
Bug 1960129: add smoke tests about VM pages and actions

rexagod committed:
Bug 1959041: "Troubleshoot" link doesn't exist after scaling down OSD pod.

Replaced duplicate alert "CephOSDDiskUnavailable" to "CephOSDDiskNotResponding" for
"console.alert-action" type.

metalice committed:
Bug 1916089: added a check on vmiProp

Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

openshift-merge-robot committed:
Merge pull request #8689 from GowthamShanmugam/fix_pool

Bug 1946243: Fix pool creation timout issue when PG count is limit is…

kyoto committed:
Monitoring dashboards: Remove the "Grafana UI" link

The Grafana UI is still accessible via its route, but some dashboards
are only available on Console's Monitoring > Dashboards page, so
generally people should go there instead.

avivtur committed:
Bug 1959550: Overly generic CSS rules for dd and dt elements breaks styling elsewhere in console

Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=1959550
Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

metalice committed:
Bug 1920307: changed comparison type

Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

openshift-merge-robot committed:
Merge pull request #8794 from rottencandy/story/devfile-ext

Add devfile samples to samples catalog

openshift-merge-robot committed:
Merge pull request #8892 from andrewballantyne/fix-pipeline-tests

Fix node-v15 pipeline test failures

metalice committed:
Bug 1951486: replace metrics queries

Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

avivtur committed:
Bug 1926776: "Template support" modal appears when select the RHEL6 common template

Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=1926776
Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

SanjalKatiyar committed:
Fix Uncaught TypeError for OCS internal mode

openshift-merge-robot committed:
Merge pull request #8884 from gajanan-more/ODC-5576-topology-list-view-automation

Topology List View Automation

openshift-merge-robot committed:
Merge pull request #8898 from rexagod/1959041

Bug 1959041: "Troubleshoot" link doesn't exist after scaling down OSD pod

karthikjeeyar committed:
Fix pipelinerun and taskrun log snippet message for timeout issue

rohitkrai03 committed:
Updated helm chart annotations namespace

openshift-merge-robot committed:
Merge pull request #8893 from TheRealJon/1959519

Bug 1959519: Fix typo in OperandDetails components

rohitkrai03 committed:
Add helm config export in oc-environment script

makambalaji committed:
updated regression scripts for add-flow
updated gherkin scripts scenario codes for polarion management purpose

rebeccaalpert committed:
Add Sprint 199 translations

Uploading ko, zh-cn, and ja translations from Globalization team.

dtaylor113 committed:
CONSOLE-2847: Cypress: port protractor Filtering tests

openshift-merge-robot committed:
Merge pull request #8872 from sedroche/console-with-monitoring

Bug 1959916: Make console works against api proxy

openshift-merge-robot committed:
Merge pull request #8812 from invincibleJai/pr-8794-5729-2

adds devfile provider for dev catalog

openshift-merge-robot committed:
Merge pull request #8819 from rebeccaalpert/memsource-automation

CONSOLE-2508: Memsource automation

openshift-merge-robot committed:
Merge pull request #8851 from debsmita1/node-list-row-height

Bug 1935814: Fix row heights when additional columns in the pod & node list page have long text

Commit:
Adding support to consume latest stable images of ocs operator

- Plugin tests can consume the latest stable image
- We adopted the methodology of linking the private repo secret to all service accounts in order to be consumed by pods.
- The secret is created in `openshift-stoarge` namespace to be consumed by all pods for pulling their bundles
- The secret is created in `openshift-marketplace` in order to create a custom catalog source

Fixes https://issues.redhat.com/browse/RHSTOR-1835

Signed-off-by: Afreen Rahman <afrahman@redhat.com>

yapei committed:
Bug 1960089: CamelCase ImageStream related pages and breadcrumb

avivtur committed:
Followup to PR #8843

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

openshift-merge-robot committed:
Merge pull request #8913 from dtaylor113/protractor-filtering-to-cypress

CONSOLE-2847: Cypress: port protractor Filtering tests

yaacov committed:
bypass failing tests

avivtur committed:
Bug 1908169: The example of Import URL is "Fedora cloud image list" for all templates.

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

yaacov committed:
override template apiversion

kyoto committed:
Monitoring dashboards: Support colored text for single values

Support the `thresholds` object, which specifies colors for single stats
depending on their value.

Color names are defined by the Grafana dashboard definition format. They
are mapped to PatternFly chart colors to be consistent with the graph
colors on the same page.

Also adds support for `gauge` panels, but just renders them as single
values (with support for threshold colors).

pcbailey committed:
Fix crashing of VM Templates page

dtaylor113 committed:
CONSOLE-2855: Cypress: port protractor Storage tests

openshift-merge-robot committed:
Merge pull request #8874 from yaacov/v1-api

Bug 1958811: KubeVirt v1 GA api

openshift-merge-robot committed:
Merge pull request #8804 from reginapizza/gitops-issue-878

GitOps- Convert list card view to table view

openshift-merge-robot committed:
Merge pull request #8921 from kyoto/monitoring-dashboards-colors

Bug 1940318: Monitoring dashboards: Support colored text for single values

openshift-merge-robot committed:
Merge pull request #8877 from kdoberst/Bug-1930007-Allow-multiple-select-on-the-Search-and-events-page

Bug 1930007: Allow multiple selection on resources drop down

openshift-merge-robot committed:
Merge pull request #8905 from avivtur/bz_1926776

Bug 1926776: "Template support" modal appears when select the RHEL6 common template

andrewballantyne committed:
Fix CopyToClipboard tests that lacked mocks

openshift-merge-robot committed:
Merge pull request #8875 from yaacov/depracte-isvmready

Bug 1958868: Depracte isvmready

openshift-merge-robot committed:
Merge pull request #8886 from metalice/vm-status-component-change

Added missing icons to VM states and removing popup when no actual message available

openshift-merge-robot committed:
Merge pull request #8900 from kyoto/monitoring-dashboards-remove-grafana-link

Bug 1960059: Monitoring dashboards: Remove the "Grafana UI" link

openshift-merge-robot committed:
Merge pull request #8918 from avivtur/followupto_PR8843

Followup to PR #8843

keithchong committed:
GitOps: Add Sync Status and Latest Deployment Time to List Page (751)

Signed-off-by: Keith Chong <kykchong@redhat.com>

openshift-merge-robot committed:
Merge pull request #8848 from abhinandan13jan/dndIcon

update dnd icon

openshift-merge-robot committed:
Merge pull request #8880 from afreen23/master

Bug 1957756: Fix key value in replacement object

openshift-merge-robot committed:
Merge pull request #8901 from avivtur/bz_1959550

Bug 1959550: Overly generic CSS rules for dd and dt elements breaks styling elsewhere in console

openshift-merge-robot committed:
Merge pull request #8902 from metalice/bz-1920307-vm-not-running-should-be-guest-agent-not-installed

Bug 1920307: changed comparison type

openshift-merge-robot committed:
Merge pull request #8904 from metalice/bz-1951486-replace-query-metrics

Bug 1951486: replace metrics queries

openshift-merge-robot committed:
Merge pull request #8899 from metalice/bz-1916089-vmi-nic-tab

Bug 1958753: added a check on vmiProp

debsmita1 committed:
fix helm rollback form footer alignment

openshift-merge-robot committed:
Merge pull request #8836 from makambalaji/odc-5803-epic-automation-for-workspaces-support

ODC-5803-epic automation for workspaces

Commit:
Gherkin: QuickStarts-enhanced-capabilities-ODC-5010

invincibleJai committed:
fixes monitoring nav route for dev perspective

pcbailey committed:
Default to current namespace for non-common templates

rebeccaalpert committed:
Bug 1942117: Add Storage page

Some parts of the Add Storage page were missed during the first round of i18n.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1942117

openshift-merge-robot committed:
Merge pull request #8325 from makambalaji/odc-5364-consolidate-index-js-files

ODC-5364-Consolidate plugins/index.js file in cypress framework

spadgett committed:
Bug 1960683: Fix hot loop in global config page

`oauthMenuItems` was initialized to a different value on every render,
and it was included in the dependency list of the `useEffect` hook. This
caused the hook to hot loop. Initialize `oauthMenuItems` inside
`useEffect` instead since it's only used in that block. Also remove
`errors` from the dependency list since it wasn't referenced inside
the hook.

invincibleJai committed:
fixes issue with create menuitems on serverless, pipelines nav options

openshift-merge-robot committed:
Merge pull request #8894 from andrewballantyne/pipeline-i18n-refactor

Pipeline i18n refactor

openshift-merge-robot committed:
Merge pull request #8929 from invincibleJai/odc-5836

fixes monitoring nav route for dev perspective

openshift-merge-robot committed:
Merge pull request #8906 from SanjalKatiyar/fix_typeerror_internal_wizard

Fix Uncaught TypeError for OCS Internal mode

openshift-merge-robot committed:
Merge pull request #8917 from yapei/camelcase-is

Bug 1960089: CamelCase ImageStream related pages and breadcrumb

openshift-merge-robot committed:
Merge pull request #8922 from pcbailey/fix-templates-page-crash

Fix crashing of VM Templates page

openshift-merge-robot committed:
Merge pull request #8932 from rebeccaalpert/add-storage

Bug 1942117: i18n Add Storage page

openshift-merge-robot committed:
Merge pull request #8931 from keithchong/751-879-SyncStatusAndLatestDeploymentTime

GitOps: Add Sync Status and Latest Deployment Time to List Page (751)

openshift-merge-robot committed:
Merge pull request #8860 from abhinandan13jan/taskruntab

Bug 1954105: Update Taskrun tab to make namespace based calls

sangupta1 committed:
moved Details component into alert-details.tsx file

gajanan-more committed:
Topology - Delete Workload

openshift-merge-robot committed:
Merge pull request #8942 from gajanan-more/ODC-5582-delete-workload

Topology - Delete Workload

kyoto committed:
Monitoring dashboards: Fix bug where panels could be duplicated

We should clone the panel so that the `row.panels.push` action isn't
altering the original panel's rows.

openshift-merge-robot committed:
Merge pull request #8869 from jotak/fix-doc-link

Bug 1961064: Fix documentation link to network policies

openshift-merge-robot committed:
Merge pull request #8938 from invincibleJai/odc-5842

fixes issue with create menuitems on serverless, pipelines nav options

openshift-merge-robot committed:
Merge pull request #8879 from rottencandy/fix/git-detect

Fix gitlab & bitbucket builder image detection

Commit:
Increase backing store deletion timeout  from 30s to 1 min.

The deletion is flaky at times. We want the cleanup() to finish before moving on next step.
Currently 30s default time lapse is not enough.

Signed-off-by: Afreen Rahman <afrahman@redhat.com>

sangupta1 committed:
performed rebase with upstream

divyanshiGupta committed:
Hide object updated alert on reload

Commit:
gherkin to add certification filter and certified badge in helm catalog-ODC-5840

openshift-merge-robot committed:
Merge pull request #8720 from rebeccaalpert/sprint199-translations

CONSOLE-2527: Sprint 198 translations

openshift-merge-robot committed:
Merge pull request #8943 from kyoto/monitoring-dashboards-fix-duplicate-panels

Bug 1960531: Monitoring dashboards: Fix bug where panels could be duplicated

openshift-merge-robot committed:
Merge pull request #8908 from rohitkrai03/helm-annotations-fix

Updated helm chart annotations namespace

openshift-merge-robot committed:
Merge pull request #8933 from debsmita1/helm-rollback-formfooter

fix helm rollback form footer alignment

openshift-merge-robot committed:
Merge pull request #8946 from divyanshiGupta/ODC-5847

Fix to hide object updated alert on reload in yaml view

andrewballantyne committed:
Remove the broken Devfile Sample for BuildConfigs

openshift-merge-robot committed:
Merge pull request #8820 from dtaylor113/test-prow-wait-for-auth-operator

Bug 1897621: Auth test.Login times out - Wait for Cluster Auth Operator to start and finish Progressing 'test' IDP before e2e tests starts

openshift-merge-robot committed:
Merge pull request #8923 from dtaylor113/protractor-storage-to-cypress

CONSOLE-2855: Cypress: port protractor Storage tests

openshift-merge-robot committed:
Merge pull request #8934 from spadgett/fix-global-config-hotloop

Bug 1960683: Fix hot loop in global config page

kyoto committed:
Monitoring: Remove "Alertmanager UI" link

The Alertmanager UI is still accessible via its route, but all necessary
information should be available through the Console admin UI.

SanjalKatiyar committed:
fix_used_capacity_breakdown_pods_status

openshift-merge-robot committed:
Merge pull request #8924 from afreen23/add-custom-catalog

Adding support to consume latest stable images of ocs operator

openshift-merge-robot committed:
Merge pull request #8948 from andrewballantyne/remove-buildconfig-devfile-sample

Bug 1961336: Remove the broken Devfile Sample for BuildConfigs

metalice committed:
Bug 1950916: added warning message when vm is paushed on consloe page and the option to pause a machine from kebab menu

Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

vbnrh committed:
Enables text input for volume count in backingstore page

Signed-off-by: Vineet Badrinath <vbadrina@redhat.com>

openshift-merge-robot committed:
Merge pull request #8957 from metalice/bz-1950916-console-connect-error-message

Bug 1950916: added warning message when vm is paused on console page and the option to pause a machine from kebab menu

yaacov committed:
re enable customize boot source tests

kdoberst committed:
Add i18n to new route dropdown placeholders

Add 18n for following placeholder text on create route:
- Select a service
- Select target port
- Select termination type
- Select insecure traffic type

openshift-merge-robot committed:
Merge pull request #8428 from pcbailey/add-custom-temp-from-common-temp-test

Add tests for creating custom templates from common templates

openshift-merge-robot committed:
Merge pull request #8856 from pcbailey/replace-popover-with-fieldlevelhelp-for-form-help

Convert form field help uses of Popover to FieldLevelHelp

sg00dwin committed:
Switch to double quotes
fixes bug https://bugzilla.redhat.com/show_bug.cgi?id=1961356

andrewballantyne committed:
Allow ManagedFields Code Folding on Form/YAML Switcher

debsmita1 committed:
fix dev-catalog type tooltip

openshift-merge-robot committed:
Merge pull request #8955 from SanjalKatiyar/used_capacity_breakdown_status

Bug 1954460: The status of "Used Capacity Breakdown [Pods]" is "Not available"

Commit:
Gherkin for making deployment/deployment-config serverless:ODC-5422

openshift-merge-robot committed:
Merge pull request #8199 from dtaylor113/acm-perspective-switcher

CONSOLE-2506: Added 'Advanced Cluster Management' to Perspective Switcher

openshift-merge-robot committed:
Merge pull request #8825 from SanjalKatiyar/add_external_fsname_text_box

Bug 1924729: Create Storageclass for CephFS provisioner assumes incorrect default FSName in external mode

openshift-merge-robot committed:
Merge pull request #8868 from karthikjeeyar/fix-metrics-tooltip

Fix tooltip on pipelineRunCount and TaskRunDuration graphs based of P…

openshift-merge-robot committed:
Merge pull request #8882 from TheRealJon/1943238

Bug 1943238: Display olm conditions descriptors in their own page section on operand details pages

openshift-merge-robot committed:
Merge pull request #8907 from karthikjeeyar/fix-timeout-log-snippet

Fix pipelinerun and taskrun log snippet message for timeout issue

openshift-merge-robot committed:
Merge pull request #8952 from kyoto/monitoring-remove-alertmanager-link

Bug 1961452: Monitoring: Remove "Alertmanager UI" link

openshift-merge-robot committed:
Merge pull request #8961 from kdoberst/Bug-1961466-Dropdown-text-translation

Bug 1961466: Some dropdown placeholder text on route creation page is not translated

openshift-merge-robot committed:
Merge pull request #8963 from sg00dwin/webhooksecretkey-bug1961356

Bug 1961356: Switch to double quotes in translation

DaoDaoNoCode committed:
remove history listener during unmounting period

gouyang committed:
remove quick starts check on Add page

openshift-merge-robot committed:
Merge pull request #8964 from andrewballantyne/fix-yaml-code-folding

Bug 1932472: Allow ManagedFields Code Folding on Form/YAML Switcher

kyoto committed:
Monitoring dashboards: Allow variable dropdowns to default to "All"

openshift-merge-robot committed:
Merge pull request #8912 from rebeccaalpert/sprint199-new-translations

Bug 1961878: Add Sprint 199 translations

openshift-merge-robot committed:
Merge pull request #8970 from gouyang/remove_quickstarts

remove quick starts check on Add page

christoph-jerolimov committed:
Add getting started section to the add page

sangupta1 committed:
used type K8sResourceKind and made Details as functional component and other minor fixes

bipuladh committed:
Remove guards from Core OCS Operator API Pages

yaacov committed:
remove extra space in string

sangupta1 committed:
fixing the frontend test

openshift-merge-robot committed:
Merge pull request #8939 from pcbailey/default-to-current-ns-for-non-common-templates

Bug 1917328: Default to current namespace for non-common templates

jotak committed:
Bug 1962153: fix VolumeSnapshot routes

- Fix routes so that they're not ambiguous
- Fix Edit YAML link in case we're coming from a PVC page
- Simplify create-volume-snapshot:
    - async loading of kind data shouldn't be necessary
    - rename "resourceName" to "pvcName", more explicit
    - absence/presence of pvcName props is sufficient to decide conditional
      display of the PVC dropdown

dankenigsberg committed:
virt: use the new rhel8 image as an example

For downtream it would be better than Fedora

Signed-off-by: Dan Kenigsberg <danken@redhat.com>

makambalaji committed:
updated helm automation scripts

rottencandy committed:
hide taskrun edit actions in taskrun list view

openshift-merge-robot committed:
Merge pull request #8888 from rawagner/details_kind

Bug 1961393: Use K8sModel as kindObj if available

openshift-merge-robot committed:
Merge pull request #8911 from makambalaji/odc-5807-create-or-fix-add-flow-scripts

ODC- 5807-Update add flow automation scripts

openshift-merge-robot committed:
Merge pull request #8959 from yaacov/e2e-re-enable-customize-boot-source-tests

Re enable customize boot source tests

openshift-merge-robot committed:
Merge pull request #8975 from kyoto/monitoring-dashboards-support-default-all

Bug 1962062: Monitoring dashboards: Allow variable dropdowns to default to "All"

rohitkrai03 committed:
Fix duplicate helm charts not showing up in catalog

openshift-merge-robot committed:
Merge pull request #8748 from vbnrh/fix-number-of-volume-input

Bug 1949889: Enables text input for volume count in backingstore page

openshift-merge-robot committed:
Merge pull request #8960 from rottencandy/fix/taskrun-action

Bug 1961675: Hide TaskRun edit actions for rows in Pipelinerun's TaskRun tab

sangupta1 committed:
added translation for alert logs

openshift-merge-robot committed:
Merge pull request #8976 from bipuladh/core-api-changes

Remove guards from Core OCS Operator API Pages

openshift-merge-robot committed:
Merge pull request #8977 from yaacov/kubevirt-plugin-fix-typo

Bug 1961363: remove extra space in string

openshift-merge-robot committed:
Merge pull request #8980 from dankenigsberg/rhel8-guest-image-as-example

Bug 1962256: virt: use the new rhel8 image as an example

openshift-merge-robot committed:
Merge pull request #8983 from rohitkrai03/helm-duplicate-fix

Fix duplicate helm charts not showing up in catalog

openshift-merge-robot committed:
Merge pull request #8867 from nemesis09/add-page-masonry-layout

Redesign add page to follow masonry layout

kyoto committed:
Monitoring dashboards: Always allow custom format for tooltip entries

The `legendFormat` function is used for formatting the legend entries
both below the graph and in the graph tooltip. However, we were ignoring
it when `panel.legend.show` isn't `true`. This changes the tooltips to
always use `legendFormat` when it is provided.

yaacov committed:
Use RHEL8 image in new from yaml

Signed-off-by: yaacov <kobi.zamir@gmail.com>

jotak committed:
Merge volumesnapshot similar routes use query params

and rename variable

openshift-merge-robot committed:
Merge pull request #8916 from gouyang/smoke_test

Bug 1960129: Add smoke tests to verify pages are able to load

gajanan-more committed:
Topology-Edit Workload and Resource Type

sangupta1 committed:
deleted details-page.json

openshift-merge-robot committed:
Merge pull request #8920 from avivtur/bz_1908169

Bug 1908169: The example of Import URL is "Fedora cloud image list" for all templates.

openshift-merge-robot committed:
Merge pull request #8965 from debsmita1/dev-catalog-tooltip

fix dev-catalog type tooltip

gkarat committed:
When no issues, show Insights widget content

The patch solves the bug when no recommendations hit a cluster,
and the widget had no content inside.

rohitkrai03 committed:
Fix golang detection regex to only match .go files

rohitkrai03 committed:
Fix quick start prerequisites getting truncated

openshift-merge-robot committed:
Merge pull request #8989 from gkarat/insights-widget-no-issues-no-content-bug

Bug 1949338: show the content of Insights widget when there are 0 recommendations for cluster

rohitkrai03 committed:
Auto focus first input field of import forms

openshift-merge-robot committed:
Merge pull request #8950 from jerolimov/odc-5560

Add getting started section to the add page in developer perspective

openshift-merge-robot committed:
Merge pull request #8992 from rohitkrai03/golang-detection-fix

Fix golang detection regex to only match .go files

openshift-merge-robot committed:
Merge pull request #8664 from nemesis09/pipelinerun-vis-status-pending

pipelinerun visualisation for status PipelineRunPending

openshift-merge-robot committed:
Merge pull request #8985 from yaacov/use-shiny-new-rhel8-image-in-yaml

Bug 1962256: Use shiny new rhel8 image in yaml

openshift-merge-robot committed:
Merge pull request #8986 from kyoto/query-browser-add-show-legend-flag

Bug 1960534: Monitoring dashboards: Always allow custom format for tooltip entries

openshift-merge-robot committed:
Merge pull request #8972 from sanketpathak/gherkin-make-workload-serverless

Gherkin for making deployment/deployment-config serverless

openshift-merge-robot committed:
Merge pull request #8988 from gajanan-more/ODC-5583-edit-application

Topology-Edit Workload and Resource Type

rebeccaalpert committed:
Bug 1963159: Add Sprint 200 translations

Added Sprint 200 translations and removed unused language files from before Zac's consolidation effort.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1963159

openshift-merge-robot committed:
Merge pull request #8910 from rohitkrai03/helm-local-setup-fix

Add helm config export in oc-environment script

openshift-merge-robot committed:
Merge pull request #8981 from makambalaji/odc-5854-helm-scripts-update

ODC-5854-updated helm automation scripts

rohitkrai03 committed:
Fix project deletion issue and reset active namespace

openshift-merge-robot committed:
Merge pull request #8891 from sanketpathak/gherkin-Support_for_Managed_Service-ODC-5175

Gherkin: Implement the support needed for managed kafka connection, Visualisation of the serverless function in topology and Connection from kafka source to kafka connection

openshift-merge-robot committed:
Merge pull request #8994 from rohitkrai03/qs-tile-fix

Bug 1961829: Fix quick start prerequisites getting truncated

rebeccaalpert committed:
Bug 1963219: i18n ImageStreamTag page

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1963219

florkbr committed:
Bug 1955803: OperatorHub duplicate item details and empty infraFeatures

As pointed out by @yapei on the support release-4.7 PR, there are
two remaining issues with the infrastructure features filter and display
values:

1. There are duplicate items listed under `Infrastructure Features` on the operator item details page
2. For operators with no infrastructure features (such as `Kubestone`), we need to hide the section

Follow up to original PR in master: https://github.com/openshift/console/pull/8844

https://bugzilla.redhat.com/show_bug.cgi?id=1955803

openshift-merge-robot committed:
Merge pull request #8979 from jotak/fix-volumesnapshot-routes

Bug 1962153: fix VolumeSnapshot routes

openshift-merge-robot committed:
Merge pull request #9003 from rohitkrai03/delete-project-fix

Bug 1945312: Fix project deletion issue and reset active namespace

openshift-merge-robot committed:
Merge pull request #8996 from rohitkrai03/auto-focus-field

Auto focus first input field of import forms

openshift-merge-robot committed:
Merge pull request #9005 from rebeccaalpert/imagestreamtag-i18n

Bug 1963219: i18n ImageStreamTag page

christianvogt committed:
adjust styles of add page and getting started to align with UX

openshift-merge-robot committed:
Merge pull request #8969 from DaoDaoNoCode/bug-1961897

Bug 1961897: Remove history listener before console UI is unmounted

openshift-merge-robot committed:
Merge pull request #9006 from florkbr/1955803-bug-infrastructure-features-operator-annotation-patch

Bug 1955803: OperatorHub duplicate item details and empty infraFeatures

openshift-merge-robot committed:
Merge pull request #9004 from rebeccaalpert/sprint200

Bug 1963159: Add Sprint 200 translations

openshift-merge-robot committed:
Merge pull request #9008 from christianvogt/add-styles

adjust styles of add page and getting started to align with UX

yaacov committed:
Template support text fix

Signed-off-by: yaacov <kobi.zamir@gmail.com>

yaacov committed:
Add tooltip to creat new template kebab action item

Signed-off-by: yaacov <kobi.zamir@gmail.com>

yaacov committed:
Remove flavor and workflow when choosing a template familiy

Signed-off-by: yaacov <kobi.zamir@gmail.com>

openshift-merge-robot committed:
Merge pull request #9012 from yaacov/vm-wizard-choose-os-not-template

Bug 1963676: Vm wizard choose os not template

openshift-merge-robot committed:
Merge pull request #8591 from rexagod/1947311

Bug 1947311: Run discovery directly on nodes

openshift-merge-robot committed:
Merge pull request #8945 from sanketpathak/gherkin-Add-Helm-Chart-Catalog-filter-for-certification

Gherkin:Add certification filter and certified badge in helm catalog

rohitkrai03 committed:
Make disabled action items not clickable and focusable

yaacov committed:
Add PVC link to disk table

Signed-off-by: yaacov <kobi.zamir@gmail.com>

openshift-merge-robot committed:
Merge pull request #8927 from sanketpathak/gherkin-QuickStarts-enhanced-capabilities-ODC-5432

Gherkin: QuickStarts-enhanced-capabilities

openshift-merge-robot committed:
Merge pull request #9011 from yaacov/create-template-from-info

Bug 1963502: Add tooltip to creat new template kebab action item

rohitkrai03 committed:
Fix number spinner in  NumberSizeInput without default values

openshift-merge-robot committed:
Merge pull request #8400 from glekner/dp-sdk-yamltemplates

converting yaml-templates extension types to dynamic-plugin-sdk

openshift-merge-robot committed:
Merge pull request #9021 from yaacov/add-pvd-link-to-disk-table

Bug 1963896: Add PVC link to disk table

openshift-merge-robot committed:
Merge pull request #9018 from rohitkrai03/disabled-action-item-fix

Bug 1957982: Make disabled action items not clickable and focusable

openshift-merge-robot committed:
Merge pull request #9023 from rohitkrai03/number-spinner-fix

Fix number spinner in  NumberSizeInput without default values

openshift-merge-robot committed:
Merge pull request #9009 from yaacov/kubevirt-i18n-typo

Bug 1963211: Kubevirt i18n typo

sangupta1 committed:
rebased and fixed silenced by translation

sangupta1 committed:
Integrated Log Exploration API

sangupta1 committed:
minor changes for API
